core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -31,164 +35,182 @@ export interface QuestionBackendDict {
   'inapplicable_skill_misconception_ids': string[];
 }
 
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(
-        id, stateData, languageCode, version, linkedSkillIds,
-        inapplicableSkillMisconceptionIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-      this._inapplicableSkillMisconceptionIds = (
-        inapplicableSkillMisconceptionIds);
-    };
-
-    // ---- Instance methods ----
+export class Question {
+  _id: string;
+  _stateData: State;
+  _stateDataSchemaVersion: number;
+  _languageCode: string;
+  _version: number;
+  _linkedSkillIds: string[];
+  _inapplicableSkillMisconceptionIds: string[];
```

Comment:
Can you add `private` modifier for all these variables?

Reply:
Fixed (as per your comment below)